ngraph/core/include/ngraph/file_util.hpp

```diff
@@ -77,5 +77,12 @@ namespace ngraph
         /// \param str A null-terminated string
         /// \return A wide-char string
         NGRAPH_API std::wstring multi_byte_char_to_wstring(const char* str);
+
+        /// \brief Remove path components which would allow traversing up a directory tree
+        ///        from a base path.
+        /// \param path A path which will be santized
+        /// \return Path components to remove
+        NGRAPH_API std::string sanitize_path(const std::string& path,
```

Comment:
`sanitize_path` should take only the `path` parameter. The other parameter is not a variable, we should define the sanitize logic entirely inside the function.

Also, we can give this function a more specific name, like `prevent_absolute_path` or something similar.

Reply:
- `to_remove` argument removed
- improved handling Windows-style cases
- I am not sure about `prevent_absolute_path` name. Paths like: ../../weights.data which are sanitized by this method are not absolute